Library/Homebrew/cask/cask_loader.rb

```diff
@@ -69,11 +69,16 @@ def self.can_load?(ref)
 
       attr_reader :token, :path
 
-      def initialize(path) # rubocop:disable Lint/MissingSuper
+      def initialize(path, tap: nil) # rubocop:disable Lint/MissingSuper
         path = Pathname(path).expand_path
 
         @token = path.basename(path.extname).to_s
+        # For files we've downloaded, they will be prefixed with `{URL MD5}--`.
+        # Remove that prefix to get the original filename.
+        @token = @token.split("--", 2).last if path.dirname == HOMEBREW_CACHE/"downloads"
```

Comment:
Perhaps given it's in formulary as well, though I'm not sure where beyond a util method?

Reply:
@Bo98 I guess it's `DownloadStrategy` that decides this logic? a `DownloadStrategy.download_split_something` method (obviously not that name and maybe not even that class) feels like it'd most make sense?

Reply:
`DownloadStrategy#resolved_basename` should return this.

Reply:
> `DownloadStrategy#resolved_basename` should return this.

It does, but is tricky to pass that result when it gets to formulary where it may reload the file in subprocesses - at which point the concept of a download strategy instance is long gone. So the logic really only works when inside the loaders themselves.

Reply:
Ok I've overhauled this by making an `api-source` cache folder which contains tap information and each file is symlinked as its original basename.

Does require some custom cleanup.rb logic, but relying on the original logic was incorrect anyway.